src/lib/dusk.rs

```diff
@@ -0,0 +1,247 @@
+        dusk::from_dusk(*val)
+    }
+}
+
+impl From<&Dusk> for f64 {
+    fn from(val: &Dusk) -> f64 {
+        (*val).into()
+    }
+}
+
+/// Lux represent Dusk in their underlying unit type
+impl From<Lux> for Dusk {
+    fn from(lux: Lux) -> Self {
+        Self(lux)
+    }
+}
+
+/// Strings are parsed as Dusk values (floats)
+impl FromStr for Dusk {
+    type Err = ParseFloatError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        f64::from_str(s).map(Dusk::from)
+    }
+}
+
+/*
+#[allow(clippy::from_over_into)]
+impl Into<f64> for Dusk {
+    fn into(self) -> f64 {
+        dusk::from_dusk(self.0)
+    }
+}
+*/
+
+/// Dusk derefs into its underlying Lux amount
+impl Deref for Dusk {
+    type Target = Lux;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// Display
+/// Let the user print stuff
+
+impl fmt::Display for Dusk {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let v: f64 = self.into();
+        f64::fmt(&v, f)
+    }
+}
+
+impl fmt::Debug for Dusk {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let v: f64 = self.into();
+        write!(f, "{}", v)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn basics() {
+        let one = Dusk::from(1.0);
+        let dec = Dusk::from(2.25);
+        assert_eq!(one, 1.0);
+        assert_eq!(dec, 2.25);
+        assert_eq!(Dusk::MIN, 0);
+        assert_eq!(Dusk::MIN, Dusk::from(0.0));
+    }
+
+    #[test]
+    fn compare_dusk() {
+        let one = Dusk::from(1.0);
+        let two = Dusk::from(2.0);
+        let dec_a = Dusk::from(0.00025);
+        let dec_b = Dusk::from(0.00190);
+        assert!(one == one);
+        assert!(one != two);
+        assert!(one < two);
+        assert!(one <= two);
+        assert!(one >= one);
+        assert!(dec_a < dec_b);
+        assert!(one > dec_b);
+    }
+
+    #[test]
+    fn ops_dusk_dusk() {
+        let one = Dusk::from(1.0);
+        let two = Dusk::from(2.0);
+        let three = Dusk::from(3.0);
+        assert_eq!(one + two, three);
+        assert_eq!(three - two, one);
+        assert_eq!(one * one, one);
+        assert_eq!(two * one, two);
+        assert_eq!(two / one, two);
+        let point_five = Dusk::from(0.5);
+        assert_eq!(one / two, point_five);
+        assert_eq!(point_five * point_five, Dusk::from(0.25))
+    }
+
+    #[test]
+    fn conversions() {
+        let my_float = 35.049;
+        let dusk: Dusk = my_float.into();
+        assert_eq!(dusk, my_float);
+        let one_dusk = 1_000_000_000u64;
+        let dusk: Dusk = one_dusk.into();
+        assert_eq!(dusk, 1.0);
+        assert_eq!(*dusk, one_dusk);
+        let dusk = Dusk::from_str("69.420").unwrap();
+        assert_eq!(dusk, 69.420);
+        let float: f64 = dusk.into();
+        assert_eq!(float, 69.420);
+        let borrowed = &Dusk(one_dusk);
+        let float: f64 = borrowed.into();
+        assert_eq!(float, 1.0);
+    }
+}
```

Comment:
You need to add also tests that ensure the `MIN` and `MAX` are taken into account. What I meant is:

1. If you try to convert negative number should panic.
2. If you try to do `Dusk::from(1.0) - Dusk::from(2.0)` should panic.
3. If you try to do `Dusk::MAX + Dusk::from(2.0)` should just return `Dusk::MAX`(I think this is the default behavior with `f64::MAX`, just double check).

You should also check that conversion from zero works as expected.

If you want to check how you write unit tests that panic, see: https://doc.rust-lang.org/book/ch11-01-writing-tests.html#checking-for-panics-with-should_panic